clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -710,37 +709,18 @@ static void populateIntHeader(SYCLIntegrationHeader &H, const StringRef Name,
     uint64_t Offset = Layout.getFieldOffset(Fld->getFieldIndex()) / 8;
 
     if (Util::isSyclAccessorType(ArgTy)) {
-      // The parameter is a SYCL accessor object - split into three
-      // parameters, so need to generate three descriptors.
-      // ... first descriptor (translated to pointer kernel parameter):
+      // The parameter is a SYCL accessor object.
+      // The Info field of the parameter descriptor for accessor contains
+      // two template parameters packed into thid integer field:
```

Comment:
I do not understand this sentence.

But great to have all this new documentation. I love reading a lot of comments while reading some code. :-)

By the way @vladimirlaz can you wait a little bit longer before merging the commits? Because of the different timezones we do not have time to just look at the commits before merging...

Reply:
Agree to wait till next working day to let everybody provide comments

Reply:
I tried to make the comment better than it was before (it though had some misprint at L714: thid -> this).
It is sad that it causes questions and needs some rephrasing/rewriting.

Would this one be better? (I could update this place sometime in future when fix this file again).

The 2nd argument of the call H.addParamDesc() below defines what is written
to the accessor parameter descriptor and its int field 'Info'.
For the template class accessor<Type, Dimensions, Mode, Target, IsPlaceHolder>
it keeps both 'Target' and 'Dimensions' packed into that int field 'Info' as:
  (Target | (Dimensions << 11)).